src/smartpeak/include/SmartPeak/core/ApplicationHandler.h

```diff
@@ -92,6 +92,16 @@ namespace SmartPeak
           return sample_group_method->getParameterSchema();
       }
 
+      std::string getDescription() const
+      {
+        if (type == RawDataMethod)
+          return raw_data_method->getDescription();
+        else if (type == SequenceSegmentMethod)
+          return seq_seg_method->getDescription();
+        else
```

Comment:
We are scoped to add in at least one additional `type` in the future so I would probably recommend using an explicit `else if` instead of the `else` here.  I would also recommend some sort of error logging or warning if the type is not recognized.